includes/convertkit-mm-functions.php

```diff
@@ -37,13 +16,18 @@ function convertkit_mm_get_option( $option_name ) {
  */
 function convertkit_mm_log( $log, $message ) {
 
-	$debug = convertkit_mm_get_option( 'debug' );
+	// Initialize settings class.
+	$settings = new ConvertKit_MM_Settings();
 
-	if ( 'on' === $debug ) {
-		$log     = fopen( CONVERTKIT_MM_PATH . '/log-' . $log . '.txt', 'a+' ); // phpcs:ignore WordPress.WP.AlternativeFunctions
-		$message = '[' . gmdate( 'd-m-Y H:i:s' ) . '] ' . $message . PHP_EOL;
-		fwrite( $log, $message ); // phpcs:ignore WordPress.WP.AlternativeFunctions
-		fclose( $log ); // phpcs:ignore WordPress.WP.AlternativeFunctions
+	// Bail if debugging isn't enabled.
+	if ( ! $settings->debug_enabled() ) {
+		return;
```

Comment:
Same comment as on the other PR - it makes more sense to log here and avoid a negative conditional.

Reply:
Same question as on the other PR - most of the Plugins have been written in this coding style. Curious why this is now an issue?